src/main/python/camdkit/framework.py

```diff
@@ -7,22 +7,188 @@
 from enum import Enum
 import dataclasses
 import re
+import importlib
 
 INT_MAX = 2147483647 # 2^31 - 1
 INT_MIN = -2147483648 # -2^31
 UINT_MAX = 4294967295 # 2^32 - 1
+UINT48_MAX = 281474976710655 # 2^48 - 1
 
 
 class Sampling(Enum):
-  STATIC = "Static"
-  REGULAR = "Regular"
+  STATIC = "Static"   # Data that does not change for a Clip or across many Frames
+  REGULAR = "Regular" # Data that appears at regular intervals in a Clip
 
 @dataclasses.dataclass
 class Dimensions:
   "Height and width of a rectangular area"
   height: numbers.Real
   width: numbers.Real
 
+@dataclasses.dataclass
+class Orientations:
+  "Horizontal and vertical measurements"
+  horizontal: numbers.Real
+  vertical: numbers.Real
+
+@dataclasses.dataclass
+class Vector3:
+  "3 doubles x,y,z to encode - for example - a location or a translation."
+  x: typing.Optional[numbers.Real] = 0.0
+  y: typing.Optional[numbers.Real] = 0.0
+  z: typing.Optional[numbers.Real] = 0.0
+
+@dataclasses.dataclass
+class Rotator3:
+  "3 doubles pan, tilt, roll to encode - for example - a camera rotation."
+  pan: typing.Optional[numbers.Real] = 0.0
```

Comment:
I recommend against picking default values and instead force the application to explicitly specify values.